src/undomanager.js

```diff
@@ -22,21 +24,20 @@ define([
         $('.fd-undo-container').append(UNDO_ALERT);
     }
 
-    function toggleAlert(undoStack, vellum) {
+    function toggleAlert(undoStack) {
         if (undoStack.length && !alertShown()) {
             createAlert();
         } else if (undoStack.length === 0 && alertShown()) {
             $('.fd-undo-delete').remove();
         }
-        if (vellum) {
-            vellum.adjustToWindow();
-        }
     }
 
     function UndoManager(form) {
         var _this = this;
         _this.undoStack = [];
-        _this.vellum = form.vellum;
+        _this.form = form;
+
+        util.eventuality(this);
```

Comment:
This is not needed since a form event is fired below. Although...

It seems like a violation of encapsulation for undo manager to fire a form event. On the other hand, it has the advantage of decoupling the form from the UI. I think this is a fine, if not the purest, solution.

FWIW here's what I originally had in mind:

form.js
```js
function Form (opts, vellum, mugTypes) {
    ...
    //make the object event aware
    util.eventuality(this);

    this.undomanager = new UndoManager();
    this.undomanager.on('undo-reset', vellum.adjustToWindow);  // this is simple, but...
    /* to really decouple the form from the UI (and use the vellum var as little as possible) we could instead
    // re-fire as a form event to be handled in the core
    this.undomanager.on('undo-reset', this.fire.bind(this));
    */
}
```
undomanager.js
```js
function UndoManager() {
    ...
    util.eventuality(this);
}
UndoManager.prototype = {
    resetUndo: function (...) {
        ...
        this.fire({type: 'undo-reset'});
    },
    ...
}
```